src/encode.cpp

```diff
@@ -51,6 +51,27 @@ llvm::cl::opt<bool> arg_use_neg_zero(
   llvm::cl::init(false),
   llvm::cl::cat(MlirTvCategory));
 
+llvm::cl::opt<string> arg_use_arg_dims("use-fn-argument-dims",
+  llvm::cl::desc(
+    "Specify the function argument to use as a reference for the "
+          "output dynamic dims"),
+  llvm::cl::value_desc("<function_name>@idx(,<function_name>@idx)*"),
+  llvm::cl::cat(MlirTvCategory)
+);
```

Comment:
You can use `cl::list` with `llvm::cl::MiscFlags::CommaSeparated`.
There will be tools in LLVM or MLIR that use this.

Reply:
Oooh I wasn't aware of `cl::CommaSeparated` flag. This should make things much easier.

Reply:
Fixed